x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileDomainIntegTests.java

```diff
@@ -249,14 +259,137 @@ public void testGetProfileByAuthenticationWillNotCheckRealmNameForFileOrNativeRe
+                    final PlainActionFuture<Profile> future = new PlainActionFuture<>();
+                    profileService.activateProfile(authentication, future);
+                    readyLatch.countDown();
+                    startLatch.await();
+                    try {
+                        final String uid = future.actionGet().uid();
+                        logger.info("create profile [{}] for authentication [{}]", uid, authentication);
+                        allUids.add(uid);
+                    } catch (VersionConflictEngineException e) {
+                        // Updating existing profile can error with version conflict. This is the current way
+                        // of handling racing in updating existing profile.
+                        // For this test, it is acceptable to either get the same profile document
+                        // or getting a version conflict for optimistic control (NOT document already exists)
+                        assertThat(e.getMessage(), containsString("version conflict, required seqNo"));
+                    }
+                } catch (Exception e) {
+                    logger.error(e);
+                    fail("caught error when creating new profile: " + e);
+                }
+            });
+            threads[i].start();
+        }
+        if (readyLatch.await(20, TimeUnit.SECONDS)) {
+            startLatch.countDown();
+            for (Thread thread : threads) {
+                thread.join();
+            }
+            // Exactly one profile is created
+            assertThat(allUids, hasSize(1));
+            final String uid = allUids.iterator().next();
+            final Profile profile1 = getProfile(uid, Set.of());
+            assertThat(profile1.uid(), equalTo(uid));
+            assertThat(profile1.user().username(), equalTo(username));
+        } else {
+            fail("Not all threads are ready after waiting");
+        }
+    }
+
+    public void testDifferentiator() {
+        String lastUid = null;
+        for (int i = 0; i < 10; i++) {
+            String currentUid = doActivateProfile(RAC_USER_NAME, TEST_PASSWORD_SECURE_STRING).uid();
+            assertThat(currentUid, endsWith("_" + i));
+            if (lastUid != null) {
+                // Base uid is identical
+                assertThat(currentUid.substring(0, currentUid.length() - 2), equalTo(lastUid.substring(0, lastUid.length() - 2)));
+            }
+            // Manually update the username to create hash collision
+            final UpdateRequest updateRequest = client().prepareUpdate(SECURITY_PROFILE_ALIAS, "profile_" + currentUid).setDoc("""
+                {
+                  "user_profile": {
+                    "user": { "username": "should-not-be-found" }
+                  }
+                }
+                """, XContentType.JSON).setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE).request();
+            client().update(updateRequest).actionGet();
```

Comment:
I would also test that the profile with the name changed, can still be activated, ie it is not as if it was removed.

Reply:
Good point. It's added.